src/components/ContactsList.js

```diff
@@ -23,7 +23,7 @@ function ContactsList(props) {
         <h2>Contacts</h2>
       </header>
       <ul className="contacts-list">
-        {contacts.map((contact, index) => {
+        {!loading && contacts.map((contact, index) => {
           const { firstName, lastName } = contact
           return (
```

Comment:
One suggestion could be to use the ternary operator here, so loading ? your loading render, otherwise your normal render.